File: project_environment_all/js_unittest/test_update_explosions.js
const GenericWorld = require('./environment_all').GenericWorld;

class GenericWorldTestCase {
    test_update_explosions() {
        const world = new GenericWorld();
        const explosion1 = new Explosion();
        explosion1.timer = 5;
        explosion1.stage = 1;
        world.explosions.push(explosion1);
        const explosion2 = new Explosion();
        explosion2.timer = 0;
        explosion2.stage = null;
        world.explosions.push(explosion2);
        const explosion3 = new Explosion();
        explosion3.timer = 2;
        explosion3.stage = 2;
        world.explosions.push(explosion3);

        world.update_explosions();

        expect(world.explosions.length).toBe(2);
        expect(world.explosions[0].timer).toBe(4);
        expect(world.explosions[0].stage).toBe(1);
        expect(world.explosions[1].timer).toBe(1);
        expect(world.explosions[1].stage).toBe(2);
    }
}

const testCase = new GenericWorldTestCase();
testCase.test_update_explosions();

File: previous_versions/v0dot3/project_environment_all/py_unittest/test_update_explosions.py
import unittest
from environment_all import GenericWorld

class GenericWorldTestCase(unittest.TestCase):
    def test_update_explosions(self):
        world = GenericWorld()
        explosion1 = Explosion()
        explosion1.timer = 5
        explosion1.stage = 1
        world.explosions.append(explosion1)
        explosion2 = Explosion()
        explosion2.timer = 0
        explosion2.stage = None
        world.explosions.append(explosion2)
        explosion3 = Explosion()
        explosion3.timer = 2
        explosion3.stage = 2
        world.explosions.append(explosion3)

        world.update_explosions()

        self.assertEqual(len(world.explosions), 2)
        self.assertEqual(world.explosions[0].timer, 4)
        self.assertEqual(world.explosions[0].stage, 1)
        self.assertEqual(world.explosions[1].timer, 1)
        self.assertEqual(world.explosions[1].stage, 2)

if __name__ == '__main__':
    unittest.main()